### PR TITLE
Fix overlays position on scroll

### PR DESCRIPTION
**Pull request by yuriy-fix**
Requires https://github.com/vaadin/vaadin-context-menu/pull/237

### CODE REVIEW

File: src/vaadin-menu-bar-interactions-mixin.html [hunk 1]
```diff
@@ -198,11 +197,12 @@
     __alignOverlayPosition(e) {
       const overlay = e.target;
       const getRect = () => this._expandedButton.getBoundingClientRect();
```
Comment (tomivirkki) [inline src/vaadin-menu-bar-interactions-mixin.html]: What is the function still needed for?

Comment (yuriy-fix) [reply src/vaadin-menu-bar-interactions-mixin.html]: Removed.

File: src/vaadin-menu-bar-interactions-mixin.html [hunk 2]
```diff
@@ -373,6 +374,7 @@
     }
 
     _close(restoreFocus) {
+      this.style.pointerEvents = '';
```
[the remark this answers is not in the document]
Comment (yuriy-fix) [reply src/vaadin-menu-bar-interactions-mixin.html]: The problem with scheduling was on `open`. Fixed now.

File: src/vaadin-menu-bar-interactions-mixin.html [hunk 3]
```diff
@@ -200,9 +200,11 @@
       const getRect = () => this._expandedButton.getBoundingClientRect();
       if (overlay.hasAttribute('bottom-aligned')) {
         overlay.style.bottom = parseInt(getComputedStyle(overlay).bottom) + getRect().height + 'px';
+        this._subMenu.__y.eventOffset -= getRect().height;
```
[the remark this answers is not in the document]
Comment (yuriy-fix) [reply src/vaadin-menu-bar-interactions-mixin.html]: Updated.

File: src/vaadin-menu-bar-interactions-mixin.html [hunk 4]
```diff
@@ -279,6 +281,7 @@
     }
 
     __openSubMenu(button, event, options = {}) {
+      this.style.pointerEvents = 'auto';
```
[the remark this answers is not in the document]
Comment (yuriy-fix) [reply src/vaadin-menu-bar-interactions-mixin.html]: Removed on `close` and test added.

File: src/vaadin-menu-bar-interactions-mixin.html [hunk 5]
```diff
@@ -53,7 +53,7 @@
       this._subMenu.addEventListener('close-all-menus', this.__onEscapeClose.bind(this));
 
       const overlay = this._subMenu.$.overlay;
-      overlay.modeless = true;
+      overlay.modeless = false;
```
[the remark this answers is not in the document]
Comment (yuriy-fix) [reply src/vaadin-menu-bar-interactions-mixin.html]: You are right! Removed.

File: src/vaadin-menu-bar-interactions-mixin.html [hunk 6]
```diff
@@ -87,7 +86,7 @@
     }
 
     _getButtonFromEvent(e) {
-      return Array.from(e.composedPath()).filter(el => el.item)[0];
+      return Array.from(e.composedPath()).filter(el => el.localName === 'vaadin-menu-bar-button')[0];
```
Comment (tomivirkki) [inline src/vaadin-menu-bar-interactions-mixin.html]: Why was this changed? All the tests pass fine without this change.

File: src/vaadin-menu-bar-interactions-mixin.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (tomivirkki) [reply src/vaadin-menu-bar-interactions-mixin.html]: Needs a test to verify the fix

File: src/vaadin-menu-bar-interactions-mixin.html (hunk 6 above)
Comment (yuriy-fix) [reply src/vaadin-menu-bar-interactions-mixin.html]: Test is failing on IE11 without this update.

File: src/vaadin-menu-bar-interactions-mixin.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (yuriy-fix) [reply src/vaadin-menu-bar-interactions-mixin.html]: Test is updated to catch the proper timing.
